### PR TITLE
add `require_all` (close #417)

### LINKED ISSUE #417: Proposal: Add an option to require all defined fields of a schema
There are four cases which I think should be covered with validator instance arguments:
* Allow known missed, forbid unknown present (default behavior right now)
* Allow known missed, allow unknown present (`allow_unknown=True`)
* Forbid known missed, forbid unknown present (no way at validator creation time)
* Forbid known missed, allow unkown present (no way at validator creation time)

Cerberus have `allow_unknown` option for validator, but have no `allow_known_missed` / `forbid_known_missed` or something like this. As [stackoverflow](https://stackoverflow.com/questions/45935812/have-required-set-to-true-by-default-in-cerberus) question said, it is pretty annoying to set `required: True` for all fields, cause they can be nested and schema looks noisy and dirty.

### PR DESCRIPTION
**Pull request by pohmelie**
#417 
#421

### CODE REVIEW

File: docs/usage.rst [hunk 1]
```diff
@@ -162,6 +162,46 @@ mapping that is checked against the :ref:`schema <schema_dict-rule>` rule:
    ``allow_unknown`` can also be set to a validation schema.
 
 
+.. _requiring-all:
+
+Requiring all
+-------------
+See also :ref:`this paragraph <require_all>`, :ref:`this paragraph <required>`.
```
Comment (funkyfuture) [inline docs/usage.rst]: i don't think this is needed...

File: docs/usage.rst [hunk 2]
```diff
@@ -162,6 +162,46 @@ mapping that is checked against the :ref:`schema <schema_dict-rule>` rule:
...
+
+By default any keys defined in the schema are not required.
+However, you can require all document keys pairs by setting
+``require_all`` to ``True`` at validator initialization (``v = Validator(…, require_all=True)``)
+or change it latter via attribute access (``v.require_all = True``).
+``require_all`` can also be set as rule to configure a validator for a subdocument
```
Comment (funkyfuture) [inline docs/usage.rst]: when `require_all` (the rule) is linked here.

File: docs/validation-rules.rst [hunk 3]
```diff
@@ -679,6 +690,10 @@ unless :meth:`~cerberus.Validator.validate` is called with ``update=True``:
     >>> v.validate(document, update=True)
     True
 
+.. note::
+
+   To define all fields of a document as required see :ref:`this paragraph <requiring-all>`.
```
Comment (funkyfuture) [inline docs/validation-rules.rst]: 'paragraph' -> 'section about the available options' !?

File: docs/usage.rst (hunk 2 above)
Comment (pohmelie) [reply docs/usage.rst]: don't get how this should be changed.
```
``require_all`` can also be set as rule to configure a validator for a subdocument
that is checked against the :ref:`schema <schema_dict-rule>` rule:
```
→
```
``require_all`` can also be set as rule to configure a validator for a subdocument
when ``require_all`` (the rule) is linked here.
```
?

Comment (funkyfuture) [reply docs/usage.rst]: i mean this:

```
``require_all`` can also be set :ref:`as rule <require_all>` to configure …
```

**Comment by funkyfuture** [comment]
thanks a lot!

**Comment by pohmelie** [comment]
@funkyfuture thanks for help! :wink:

**Comment by nicolaiarocci** [comment]
thanks and sorry for the delay

**Pull request merged**